### ISSUE TITLE
[bug] FooterExtendedNavList sections are not accessible

### ISSUE THREAD

**Issue by haworku** [trussworks/react-uswds#281]
**Describe the bug**
`FooterExtendedNavList` turns into a collapsible list on mobile view. That list is not accessible by keyboard tabs.

`section` is not a true interactive element and we should add better aria support for the expected behavior. Specifically `role` and `aria-pressed` attributes should be added.  More context [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role).  We may also have to fiddle around to get keyboard interactions working, not sure.

**Related Issues/PRS*
#271 surfaced this again

**To Reproduce**
1. Go into mobile view on Footer and try to tab through the nav list

**Expected behavior**
The navigation items can be collapsed and expanded by keyboard tabs and are announced by screenreader. 

**More Context**
Originally used `<section />` elements rather than buttons in following with the current uswds spec.

**Comment by haworku** [comment]
Let's close this issue since USWDS Footer also doesn't handle this yet. Not entirely sure what the path forward here should be since its also mobile specific - so would like to follow their lead.
